test_scripts/Smoke/Resumption/ATF_Resumption_App_Unregister_itself.lua

```diff
@@ -0,0 +1,76 @@
+--  Requirement summary:
+--  [Data Resumption] Application data must not be resumed
+--
+--  Description:
+--  Check that no resumption occurs if App unregister itself gracefully.
+
+--  1. Used precondition
+--  App is registered and activated on HMI.
+
+--  2. Performed steps
+--  Exit from SPT
+--  Start SPT again, Find Apps
+--
+--  Expected behavior:
+--  1. SPT sends UnregisterAppInterface and EndSession to SDL. 
+--     SPT register in usual way, no resumption occurs
+
+-- [[ Required Shared Libraries ]]
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local commonStepsResumption = require('test_scripts/Smoke/Resumption/commonStepsResumption')
+
+--[[ General Settings for configuration ]]
+Test = require('user_modules/dummy_connecttest')
+require('cardinalities')
+require('user_modules/AppTypes')
+
+-- [[Local variables]]
+local default_app_params = config.application1.registerAppInterfaceParams
+local default_app = nil -- will be initialized after application registration
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+commonSteps:DeletePolicyTable()
+commonSteps:DeleteLogsFiles()
+
+function Test:Start_SDL_With_One_Activated_App()
+  self:runSDL()
+  commonFunctions:waitForSDLStart(self):Do(function()
+    self:initHMI():Do(function()
+      commonFunctions:userPrint(35, "HMI initialized")
+      self:initHMI_onReady():Do(function ()
+        commonFunctions:userPrint(35, "HMI is ready")
+        self:connectMobile():Do(function ()
+          commonFunctions:userPrint(35, "Mobile Connected")
+          self:startSession():Do(function ()
+            commonFunctions:userPrint(35, "App is registered")
+            default_app = self.applications[default_app_params.appName]
+            commonSteps:ActivateAppInSpecificLevel(self, default_app)
+            EXPECT_NOTIFICATION("OnHMIStatus", {hmiLevel = "FULL",audioStreamingState = "AUDIBLE", systemContext = "MAIN"})
+            commonFunctions:userPrint(35, "App is activated")
+          end)
+        end)
+      end)
+    end)
+  end)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("No resumption if App unregister itself")
+commonSteps:UnregisterApplication()
+
+function Test:Register_And_No_Resume_App()
+  commonStepsResumption:RegisterApp(default_app_params)
+  commonStepsResumption:ExpectNoResumeApp(default_app_params):Do(function()
+    StopSDL()
```

Comment:
@istoimenova  because we can guarantee that SDL didn't send some notification\request only if SDL was correctly shut downed and didn't sent this notification